docs/topics/db/models.txt

```diff
@@ -844,6 +844,26 @@ built-in model methods, adding new arguments. If you use ``*args,
 **kwargs`` in your method definitions, you are guaranteed that your
 code will automatically support those arguments when they are added.
 
+If you wish to update a field in the :meth:`~Model.save` method, you may also
+want to have this field automatically added to the ``update_fields`` kwarg.
+See :doc:`specifying which fields to save </ref/models/instances>`.
+This will ensure the field is saved when ``update_fields`` is specified.
+For example::
+
+
+    from django.db import models
+    from django.utils.text import slugify
+
+    class Blog(models.Model):
+        name = models.CharField(max_length=100)
+        slug = models.TextField()
+
+        def save(self, *args, update_fields, **kwargs):
+            self.slug = slugify(self.name)
+            if update_fields is not None and "name" in update_fields:
+                update_fields = set(update_fields) | {"slug"}
+            super().save(*args, **kwargs)
```

Comment:
With the current approach this is missing passing `update_fields`. So it should be:
```suggestion
            super().save(*args, **kwargs, update_fields=update_fields)
```
But, as per my previous comment, perhaps we should pass everything along separately.